internal/splunk/httprequest.go

```diff
@@ -29,6 +32,23 @@ func HandleHTTPCode(resp *http.Response) error {
 		resp.StatusCode,
 		http.StatusText(resp.StatusCode))
 
+	// Check if there is any error text returned by Splunk that we can append to the error message
+	if resp.Body != nil {
+		var jsonResponse map[string]any
+		bodyString, _ := io.ReadAll(resp.Body)
+		resp.Body = io.NopCloser(bytes.NewBuffer(bodyString))
+		unmarshalError := json.Unmarshal(bodyString, &jsonResponse)
+
+		if unmarshalError == nil {
+			responseErrorValue, ok := jsonResponse["text"]
+			if ok {
+				err = multierr.Append(err, fmt.Errorf("reason: %v", responseErrorValue))
```

Comment:
What if the json says “text”:”ok”?

Reply:
Thanks, added a test for that. If the status code is ok (like 200) it's not going to return any errors, and if for some reason there is an error status code and the message says "text": "ok", it will return: "reason: ok"